myjsonrpc.pm

```diff
@@ -38,10 +38,10 @@ sub send_json ($to_fd, $cmd) {
     $json .= "\n";
 
     confess 'myjsonprc: called on undefined file descriptor' unless defined $to_fd;
-    my $wb = syswrite($to_fd, "$json");
-    if (!$wb || $wb != length($json)) {
-        die('myjsonrpc: remote end terminated connection, stopping') if !DEBUG_JSON && $! =~ qr/Broken pipe/;
-        confess "syswrite failed: $!";
+    my $wb = syswrite($to_fd, "$json"); # wb = write buffer
+    if (!$wb || $wb != length($json)) { # is true when wb is empty or when the lenght of wb is not the entire json lenght
+        die('myjsonrpc: remote end terminated connection, stopping') if !DEBUG_JSON && $! =~ qr/Broken pipe/; # terminate the script if we are not debugging and the syserror is Broken pipe 
+        confess "syswrite failed: $!"; # write to output the error
```

Comment:
That's commenting obvious things.
Of course you might argue that it is not that obvious for a perl newbie, but then we would end up with thousands of comments in our code :)

Reply:
Yes, I've never use perl so this helps me remember what I've read. As I mention above I'll remove every comments on my code that is not strictly necessary.